File: pyguard/internal/hooks/opencode.go
package hooks

import (
	"os"
	"path/filepath"
	"strings"
)

func GenerateOpenCodePlugin(root string, pluginTemplate []byte) error {
	pyguardBin := pyguardBinary(root)
	pluginDir := filepath.Join(root, "..", ".opencode", "plugins", "pyguard")
	if err := os.MkdirAll(pluginDir, 0o755); err != nil {
		return err
	}

	tsContent := strings.ReplaceAll(string(pluginTemplate), "__BINARY__", pyguardBin)
	if err := writeHookFile(filepath.Join(pluginDir, "index.ts"), tsContent); err != nil {
		return err
	}

	opencodeContent := `{
  "$schema": "https://opencode.ai/config.json",
  "plugin": [".opencode/plugins/pyguard"]
}
`
	return writeHookFile(filepath.Join(root, "..", "opencode.json"), opencodeContent)
}

File: pyguard/skill/opencode.ts
import type { Plugin } from "@opencode-ai/plugin"

export const PknPlugin: Plugin = async ({ $ }) => {
  return {
    "file.edited": async (input: { filePath: string }) => {
      if (input.filePath.endsWith(".py")) {
        await $`__BINARY__ check`
      }
    },
  }
}
